### ISSUE TITLE
Addition of a quantity and a number does not raise an error

### ISSUE THREAD

**Issue by schymans** [sympy/sympy#12918]
If I add two quantities with inconsistent units, sympy rightly returns an error:
```
>>> from sympy.physics.units import kilogram, kelvin
>>> try: print kilogram + kelvin
>>> except Exception, error: print error
summation of quantities of incompatible dimensions
```
However, if one of the terms is a number, no error is raised:
```
>>> try: print kilogram + 1.1
>>> except Exception, error: print error
kilogram + 1.1
```

If I add two quantities, one of which does not have a dimension, an error is raised:
```
>>> from sympy.physics.units import Quantity, mass
>>> try: print Quantity('x', mass, kilogram) + Quantity('y', S.One, 1.1)
>>> except Exception, error: print error
summation of quantities of incompatible dimensions
```

If I add a quantity and a number, no error is raised:
```
>>> try: print Quantity('x', mass, kilogram) + Number(1.1)
>>> except Exception, error: print error
x + 1.1
```
I believe that this is a bug, as numbers should be treated like dimensionless quantities whenever they are used alongside with quantities.

**Comment by schymans** [comment]
Just to add to the above, during multiplication, numbers are treated like dimensionless numbers, and an error is raised in this case:
```
>>> from sympy import Eq
>>> from sympy.physics.units import Quantity, mass, volume, length, kilogram, meter
>>> density = Quantity('density', mass/volume)
>>> mass1 = Quantity('mass1', mass)
>>> mass2 = Quantity('mass2', mass)
>>> volume = Quantity('volume', length**3)
>>> eq_density = Eq(density, mass1/volume + mass2/volume)
>>> try: print eq_density.subs({volume: 1})
>>> except Exception, error: print error
summation of quantities of incompatible dimensions
```
It would be even nicer if the error message contained the terms of the summation. In any case, the user probably expects similar interpretation of numbers as dimensionless quantities if they are stand-alone summands.

**Comment by schymans** [comment]
Solved in [PR#12952](https://github.com/sympy/sympy/pull/12952).
